servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -620,6 +627,13 @@ boolean addConnection(C connection) {
                 }
 
                 final Object[] existing = previous.connections;
+                // Brute force iteration to avoid duplicates. If connections grow larger and faster lookup is required
+                // we can keep a Set for faster lookups (at the cost of more memory) as well as array.
+                for (final Object o : existing) {
+                    if (o.equals(connection)) {
+                        return true;
```

Comment:
@idelpivnitskiy - note this change. not sure connection counts are commonly high enough to justify an additional `Set` but I can add one if preferred.

Reply:
I'm fine to start with brute force and defer any optimizations here. The only concern is comparison by `equals`. If we use `equals`, the `toLoadBalancedConnection` method have to guarantee that it can distinguish between a cached duplicate of the connection and a new connection when it applies concurrency controller and the scoring. This opens a risk to cause unexpected behavior for existing users if anyone does not provide this gurantee.

Reply:
`CacheConnectionFactory` doesn't wrap the connection and it returns w/e the underlying factory provides. Not sure I follow all the control flow for `toLoadBalancedConnection` but shouldn't it just be comparing objects against the same instances of them selves when caching is enabled?

Reply:
Discussed offline and clarified ordering. After CF returns a real connection we first pass it to `toLoadBalancedConnection` and apply a concurrency controller, then apply caching layer. Therefore, comparing using default `equals` works as expected.